apps/server/src/modules/board/service/board-do-authorizable.service.spec.ts

```diff
@@ -0,0 +1,159 @@
+			const setup = () => {
+				const columnBoard = columnBoardFactory.build();
+				boardDoRepo.findById.mockResolvedValue(columnBoard);
+
+				return { columnBoardId: columnBoard.id };
+			};
+
+			it('should call the repository', async () => {
+				const { columnBoardId } = setup();
+
+				await service.findById(columnBoardId);
+
+				expect(boardDoRepo.findById).toHaveBeenCalledWith(columnBoardId, 1);
+			});
+
+			it('should return the column', async () => {
+				const { columnBoardId } = setup();
+
+				const result = await service.findById(columnBoardId);
+
+				expect(result.id).toEqual(columnBoardId);
+			});
+		});
+	});
+
+	describe('getBoardAuthorizable', () => {
+		describe('when having an empty board', () => {
+			const setup = () => {
+				const board = columnBoardFactory.build();
+				return { board };
+			};
+
+			it('should return an empty usergroup', async () => {
+				const { board } = setup();
+				boardDoRepo.findById.mockResolvedValueOnce(board);
+
+				const userGroup = await service.getBoardAuthorizable(board);
+
+				expect(userGroup.id).toEqual(board.id);
+				expect(userGroup.users.length).toEqual(0);
+			});
+		});
+
+		describe('when having a board with a teacher and some students', () => {
+			const setup = () => {
+				const roles = roleFactory.buildList(1, {});
+				const teacher = userFactory.buildWithId({ roles });
+				const substitutionTeacher = userFactory.buildWithId({ roles });
+				const students = userFactory.buildListWithId(3);
+				const course = courseFactory.buildWithId({
+					teachers: [teacher],
+					substitutionTeachers: [substitutionTeacher],
+					students,
+				});
+				const board = columnBoardFactory.build({ context: { type: BoardExternalReferenceType.Course, id: course.id } });
+				boardDoRepo.findById.mockResolvedValueOnce(board);
+				courseRepo.findById.mockResolvedValueOnce(course);
+				return {
+					board,
+					teacherId: teacher.id,
+					substitutionTeacherId: substitutionTeacher.id,
+					studentIds: students.map((s) => s.id),
+				};
+			};
+
+			it('should return the teacher and the students with correct roles', async () => {
+				const { board, teacherId, substitutionTeacherId, studentIds } = setup();
+
+				const boardDoAuthorizable = await service.getBoardAuthorizable(board);
+				const userPermissions = boardDoAuthorizable.users.reduce((map, user) => {
+					map[user.userId] = user.roles;
+					return map;
+				}, {});
+
+				expect(boardDoAuthorizable.users).toHaveLength(5);
+				expect(userPermissions[teacherId]).toEqual([BoardRoles.EDITOR]);
+				expect(userPermissions[substitutionTeacherId]).toEqual([BoardRoles.EDITOR]);
+				expect(userPermissions[studentIds[0]]).toEqual([BoardRoles.READER]);
+				expect(userPermissions[studentIds[1]]).toEqual([BoardRoles.READER]);
+				expect(userPermissions[studentIds[2]]).toEqual([BoardRoles.READER]);
+			});
+		});
+
+		describe('when trying to create a boardDoAuthorizable on a column', () => {
```

Comment:
It looks like this test is about a board object that doesn't have a `ColumnBoard` as a root.